src/huggingface_hub/commands/tag.py

```diff
@@ -0,0 +1,194 @@
+
+    def check_fields(self):
+        self.token = self.args.token if self.args.token is not None else get_token()
+        self.user = self._api.whoami(self.token)["name"]
+        if self.args.repo_type not in REPO_TYPES:
+            print("Invalid repo --repo-type")
+            exit(1)
+        if self.token is None:
+            print("Not logged in")
+            exit(1)
+
+
+class TagCreateCommand(TagCommand):
+    def run(self):
+        if self.args.message is None:
+            print("Tag message cannot be empty. Please provide a message with `--tag-message`.")
+            exit(1)
+
+        print(f"You are about to create tag {ANSI.bold(self.args.tag)} on {ANSI.bold(self.repo_id)}")
+
+        if not self.args.yes:
+            choice = input("Proceed? [Y/n] ").lower()
+            if not (choice == "" or choice == "y" or choice == "yes"):
+                print("Abort")
+                exit()
+        try:
+            self._api.create_tag(
+                repo_id=self.repo_id,
+                tag=self.args.tag,
+                tag_message=self.args.message,
+                revision=self.args.revision,
+                token=self.token,
+                exist_ok=self.args.force,
+                repo_type=self.args.repo_type,
+            )
+        except RepositoryNotFoundError:
+            print(f"Repository {ANSI.bold(self.repo_id)} not found.")
+            exit(1)
+        except RevisionNotFoundError:
+            print(f"Revision {ANSI.bold(self.args.revision)} not found.")
+            exit(1)
+        except HfHubHTTPError as e:
+            if e.response.status_code == 409:
+                print(f"Tag {ANSI.bold(self.args.tag)} already exists on {ANSI.bold(self.repo_id)}")
+                print("Use `--force` to overwrite the existing tag.")
+                exit(1)
+            raise e
+
+        print(f"Tag {ANSI.bold(self.args.tag)} created on {ANSI.bold(self.repo_id)}")
+        print("")
+
+
+class TagListCommand(TagCommand):
+    def run(self):
+        try:
+            refs = self._api.list_repo_refs(
+                repo_id=self.repo_id,
+                repo_type=self.args.repo_type,
+            )
+        except RepositoryNotFoundError:
+            print(f"Repository {ANSI.bold(self.repo_id)} not found.")
+            exit(1)
+        except HTTPError as e:
+            print(e)
+            print(ANSI.red(e.response.text))
+            exit(1)
+        if len(refs.tags) == 0:
+            print("  No tags found")
+            exit(0)
+        print("\nYour tags:")
+        for tag in refs.tags:
+            print(f"  {ANSI.bold(tag.name)}")
+        print("")
+
+
+class TagDeleteCommand(TagCommand):
+    def run(self):
+        print(f"You are about to delete tag {ANSI.bold(self.args.tag)} on {ANSI.bold(self.repo_id)}")
+
+        if not self.args.yes:
+            choice = input("Proceed? [Y/n] ").lower()
+            if not (choice == "" or choice == "y" or choice == "yes"):
```

Comment:
```suggestion
            if choice not in ("", "y", "yes"):
```

(slightly easier to read)